packages/router-core/src/router.ts

```diff
@@ -1209,6 +1210,7 @@ export class Router<
       params,
       replace,
       resetScroll,
+      state,
```

Comment:
I tried locally and on stackblitz
[forked beta v.188](https://stackblitz.com/edit/tanstack-router-bn5cgc?file=src%2Fmain.tsx)

`state` is not passing through `navigate` still...

Another point:
After the latest changes state object complies to `'true | NonNullableUpdater<LocationState> | undefined'`
![image](https://github.com/TanStack/router/assets/20629939/1a5f9c53-786d-452a-949c-af26f19f37fe)

that's not how it is on docs `// `state` is either an object of state or a function that supplies the previous state and allows you to return a new one. State is stored in the history API and can be useful for passing data between routes that you do not with to permanently store in URL search params.` 

I tried to use it as it was described on the docs - to store additional state and keep URL route simple without putting the state in search params.  I know it looks against one of the main concepts of TanStack router ;) , but as soon it's documented I tried to use it and open the PR.